x/payment/keeper/out_flow.go

```diff
@@ -0,0 +1,88 @@
+package keeper
+
+import (
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/bnb-chain/greenfield/x/payment/types"
+)
+
+// SetOutFlow set a specific OutFlow in the store from its index
+func (k Keeper) SetOutFlow(ctx sdk.Context, addr sdk.AccAddress, outFlow *types.OutFlow) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.OutFlowKeyPrefix)
+	bz, err := outFlow.Rate.Marshal()
+	if err != nil {
+		panic("should not happen")
+	}
+	store.Set(types.OutFlowKey(
+		addr,
+		outFlow.Status,
+		sdk.MustAccAddressFromHex(outFlow.ToAddress),
+	), bz)
+}
+
+// GetOutFlow get a specific OutFlow in the store from its index
+func (k Keeper) GetOutFlow(ctx sdk.Context, addr sdk.AccAddress, status types.OutFlowStatus, toAddr sdk.AccAddress) *types.OutFlow {
+	key := types.OutFlowKey(addr, status, toAddr)
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.OutFlowKeyPrefix)
+
+	value := store.Get(key)
+	if value == nil {
+		return nil
+	}
+
+	return &types.OutFlow{
+		ToAddress: toAddr.String(),
+		Rate:      types.ParseOutFlowValue(value),
+		Status:    status,
+	}
+}
+
+// GetOutFlows get OutFlows for a specific from address
+func (k Keeper) GetOutFlows(ctx sdk.Context, addr sdk.AccAddress) []types.OutFlow {
+	key := types.OutFlowKey(addr, types.OUT_FLOW_STATUS_ACTIVE, nil)
```

Comment:
Why not put the status as part of value instead of being part of the key?

Seems right now you can only get the active flow.